### DISCUSSION TITLE
List page when no token is set

### DISCUSSION THREAD

**Discussion by labrocadabro** [the-collab-lab/tcl-60-smart-shopping-list#22] [Ideas]
Looking through the upcoming issues, it doesn't look like any of the acceptance criteria specifically deal with the issue of being able to see the list page when you don't have a token set.  

I can think of two possible solutions: 
1. Hide the list link when the token is null, and if someone tries to manually visit that page, redirect them back to the home page. 
2. Combine the homepage and the list page into one. The homepage content is only ever visible when the list page isn't, and vice versa, so there's no real *need* to have two pages. (If I were designing this app myself I would have just made it an SPA. With some good UX planning, the add item functionality could also easily reside with the list.)

While I would opt for option 2 if I were making this project myself from scratch, it would require us to diverge from the AC of previous issues, so it seems to me that option 1 is the better choice here. 

Does anyone have any other suggestions for how to deal with this?

Also if @Tooba doesn't mind, I think we could add this to our PR this week. Our issue is relatively light and deals with a related issue (empty lists).

**Comment by labrocadabro** [comment]
There's also a related issue that when a token *is* set, the home link is still visible. But if you click on it, it will immediately redirect you back to the list page. It should probably be hidden when a token is set.

**Comment by RajGM** [comment]
@segdeha 
Any thoughts on this.

**Comment by Amaka202** [comment]
Thanks! Your observation is legit @labrocadabro. Our collabies in tcl-58 raised the same concern too while working on issue #3  but we talked about it during code review and they took care of it before merging their code. My suggestion is that as part of the ACs under `If a user doesn’t already have a token:` in issue #3 , we could include your first suggestion eg "If a user tries to visit the List view or AddItem view without a token saved, they should be redirected to Home page"

<img width="843" alt="Screenshot 2023-04-24 at 4 05 00 PM" src="https://user-images.githubusercontent.com/57858821/234037469-309d88b9-853b-44e1-b5f3-65a3082b5d3b.png">

And then we can suggest the "hide link" part in the `Note` section.

Your second point could work too but the home page is intended to sort of serve as our landing page and authentication/login page all in one. So that if a user already has a list and also wants to join a new existing list they can come back to the home and add their token. (This isnt within the scope of the app right now but collabies in the past have added this functionality on their own).

PS: I am not part of your team but I vote yes to you and your pairing partner taking care of it this week if you can, seeing that the token is used in fetching the users list! 😄😄

@segdeha if the suggestions are accepted I can update the repo we use to create to generate other repos for future cohorts.

**Reply by segdeha** [reply]
This sounds like a smart solve to me! Tagging in @mxmason in case they have any input before we add the change to the template repo.

**Reply by mxmason** [reply]
I like the idea of redirecting the user away from `/list` if they do not have a token! Thank you for starting this discussion, @labrocadabro!
